_posts/2023-06-21-#27748.md

```diff
@@ -10,16 +10,75 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- This PR is a follow-on to [PR 25325](https://github.com/bitcoin/bitcoin/pull/25325),
+  which we [reviewed](https://bitcoincore.reviews/25325) March 8 of this year.
+  Please review at least the notes for that review club.
+
+- It's important that the coins cache uses the amount of memory indicated by
+  the node operator via the `-dbcache` option ([the default is 450 MiB](https://github.com/bitcoin/bitcoin/blob/1ecdf6ea8fb6dc1d64f41323a1e337beb174c25b/src/txdb.h#L34)).
+  Using less memory than allowed decreases the cache _hit ratio_ (the
+  fraction of lookups that find the UTXO in the cache); using more memory
+  than specified risks crashing `bitcoind` on memory-restricted systems.
+
+- For this reason, it's important to keep an accurate accounting of the amount
+  of memory used by the cache. It doesn't have to be perfect but should be
+  fairly close.
+
+- When a program requests X bytes of dynamic memory from the C++ runtime
+  library, internally it allocates slightly more for the memory allocator's
+  metadata (overhead). In other words, logical memory is not the same
+  as physical memory.
+
+- This memory allocator metadata is somewhat complex and depends on several
+  factors, such as the machine architecture and the
+  [memory model](https://en.wikipedia.org/wiki/Memory_address#Memory_models).
+
+- When sizing the cache, we want to calculate physical memory usage, that is,
+  account for this extra allocation metadata. Unfortunately, there's no
+  library function that maps logical memory size to physical size.
+
+- To deal with that, Bitcoin Core includes a function,
+  [`MallocUsage()`](https://github.com/bitcoin/bitcoin/blob/681ecac5c2d462920cd32636eec15599a9bcf424/src/memusage.h#L50),
+  that approximates this conversion. Its argument is an allocation size,
+  and it returns the corresponding physical size.
+
+- That source file, [memusage.h](https://github.com/bitcoin/bitcoin/blob/681ecac5c2d462920cd32636eec15599a9bcf424/src/memusage.h),
+  includes many overloads of the function `DynamicUsage()` across
+  the various data types that we might be allocating somewhere
+  in the system. They all make use of `MallocUsage()`.
+
+- The pool memory resource (added by PR [#25325](https://github.com/bitcoin/bitcoin/pull/25325)) adds a new
```

Comment:
This paragraph is entirely about the previous PR, right? If so, could be good to add one more bullet point at the end describing what this PR changes/introduces?